### PR TITLE
Reset mapping before loading configuration

### LINKED ISSUE #1459: Importer: Confusing behavior when loading the configuration valid for the selected but not for the first sheet
An excel file with two sheets - `Sheet A` and `Sheet B` with different structures. A configuration stored as xlm that is valid for `Sheet B`.

- Add observed data -> Select the file.
- Select `Sheet B`
- Load configuration ->Error 
 

```
PK-Sim®

The following excel columns do not exist in the current file. 
 
'Dose [unit]
Concentration (mass) [ng/ml]' 

The corresponding mappings from the configuration will be lost. Do you want to continue?

```

I actually first have to select "Reset configuration based on selected sheet" and then load the configuration... This is not the expected behavior. I think the "Reset configuration based on selected sheet" should be performed in background when loading a configuration.

### PR DESCRIPTION
**Pull request by abdelr**
Fixes #1459 (Reset mapping before loading configuration)

### CODE REVIEW

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterPresenter.cs [hunk 1]
```diff
@@ -404,6 +404,8 @@ public void LoadConfigurationWithoutImporting()
          if (confirmDroppingOfLoadedSheets())
                return;
 
+         onResetMappingBasedOnCurrentSheet(this, null);
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ImporterPresenter.cs:407]: onResetMappingBasedOnCurrentSheet does not use either e nor null. Can I suggest to remove the argument in this mehtod, call in resetMappingBasedOnCurrentSheet and in the event registration, do (o, e)=> eesetMappingBasedOnCurrentSheet()

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterPresenterSpecs.cs [hunk 2]
```diff
@@ -429,4 +442,19 @@ public void must_filter_empty_mappings()
...
+      protected override void Because()
+      {
+         (sut as ImporterPresenterForTest).OnResetMappingBasedOnCurrentSheetInvoked = false;
+         sut.LoadConfigurationWithoutImporting();
+      }
+
+      [Observation]
+      public void must_reset_format_based_on_current_sheet()
+      {
+         (sut as ImporterPresenterForTest).OnResetMappingBasedOnCurrentSheetInvoked.ShouldBeTrue();
```
Comment (msevestre) [inline tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterPresenterSpecs.cs:457]: neat.

**Review by msevestre — CHANGES_REQUESTED** [review]
suggestion so that we do not call a method that is an event handler with weird aguments

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterPresenter.cs (hunk 1 above)
Comment (abdelr) [reply src/OSPSuite.Presentation/Presenters/Importer/ImporterPresenter.cs:407]: You are right

**Review by msevestre — APPROVED** [review]

**Pull request merged**